### ISSUE TITLE
The Fast Pipeline [ESFNet & NAS-Unet]

### ISSUE THREAD

**Issue by seekingdeep** [mit-han-lab/gan-compression#59]
@lmxyy Thank you for your hard work,
Lately, the Fast-pipeline was introduced so i did some testing to understand the strengths and limitations of the system:
1) The `cycle_gan horse2zebra` evolutionary search was able to reach the optimal configuration only after `8 iterations`.
- How is that even possible only after 8 iterations?
- Your paper states that you are able to get `65.19`, i was only able to get `fid: 76.99`, [log.zip](https://github.com/mit-han-lab/gan-compression/files/5709105/log.zip), why is that?

2) When conducting an evolutionary search on `cycle_gan horse2zebra`, only 4 samples are used, which makes it fast.
While if i use my own dataset to train a cyclegan model, it uses 100 sample in evolutionary search, which makes it very slow.
- How to set the number of samples used in the evolutionary search? [my dataset.zip](https://github.com/mit-han-lab/gan-compression/files/5709168/my.dataset.zip)

3) Another test that i did was to use full black samples for training a cycle_gan model, just to see the minimum MACs and Parameters that i can achieve. [my dataset.zip](https://github.com/mit-han-lab/gan-compression/files/5709168/my.dataset.zip)
- Can the searched model go below 2G MACs?
- if yes, then how low can it go?

4) In the task of maps and building semantic segmentation, ESFNet `ESF-interp8x` can achieve good results with only `0.527  FLOPs (G)` and `0.09 Parameters (M)`.
- Can gan-compression beat this?
https://arxiv.org/pdf/1903.12337.pdf
https://github.com/mrluin/ESFNet-Pytorch

https://ieeexplore.ieee.org/document/8681706
https://github.com/tianbaochou/NasUnet

**Comment by lmxyy** [comment]
Hi.
For the first question, it seemed that you used the old version of our model. Could you please download the latest model with [download_model.py](https://github.com/mit-han-lab/gan-compression/blob/master/scripts/download_model.py). The md5sum of the model is `34bbfec89147f331d99b433958e254e1`. We followed the setting of [once-for-all](https://github.com/mit-han-lab/once-for-all) to use 500 iterations, but the evolution often reach the optimal configuration within 40 iterations in our experiments. 

For the second questions, I do not know what do you mean by `number of samples`. The validation set of horse2zebra contains more than 100 images, but I've set the batch size to 32, so it only need 4 batches. You could limit the size of the dataset with the argument `--max_dataset_size`.

**Comment by seekingdeep** [comment]
1) i trained and searched from scratch the cyclegan horse2zebra, i was only able to get `fid: 76.99` using the default settings.
**My conclusion: i should test again to confirm the results.**

2) i was able to solve the second issue by using `--population_size 4`, now it became very fast.
This is weird, because when i searched using the original horse2zebra dataset it automatically was using population_size 4 without setting it, but when i was using my own dataset the `population_size` became 100, so i need to set it my self.
**My conclusion: it was solved using population_size, but there seems to be something weird here.**

3) Using a full black dataset "meaning all images are filled with a single color", i was able to get `1.6G MACs` and `18 fid`, these results should be closer to zero. This test is called **"Absolute Black"**, is made to see how much of the calculations is unneeded noise, and to show the model potential of using as little resources as needed. 
**My conclusion: i should retrain the initial full model so that it produces an absolute black color, then search for model**
![1](https://user-images.githubusercontent.com/71519088/102519306-5d0e5780-40a3-11eb-85aa-e54466afb781.jpg)

**Comment by seekingdeep** [comment]
3. For "Absolute Black", i was able to achieve `fid: 0.00`, it seems that `macs: 1660289024` is the minimum computational cost.
@lmxyy is it possible to achieve less that 1.6G MACs?
